tooltip dispose:removing only own event handler

david-lallement opened the pull request:
If the tooltip component is in a modal, only the listeners created by the tooltip component should to be removed in the dispose method.
For example, if we subscribe to the _**hide.bs.modal**_ event of the **_modal_** component and call the **_dispose_** method of a **_tooltip_** component inside, the _**hide.bs.modal**_ event subscription is lost.

Using of  [event namespace feature of jquery](https://api.jquery.com/event.namespace/) ensures that  only the handler created in _setListeners function is removed in dispose function.

david-lallement committed:
tooltip dispose:removing only own event handler

Johann-S commented:
Hi @david-lallement,

Can you create a live demo of the problem via [CodePen](https://codepen.io/)/[JS Bin](https://jsbin.com/) or [Stackblitz](https://stackblitz.com/) ?

Because we already use event namespace of jQuery

david-lallement commented:
Hi Johann,
a codepen to reproduce the issue : [https://codepen.io/david-lallement/pen/KjwoYx ](https://codepen.io/david-lallement/pen/KjwoYx ).

Johann-S commented:
Nice catch @david-lallement 👍 

It seems your change broke our unit tests, please can you fix that ?

david-lallement committed:
replace jquery event namespace by a named function

david-lallement commented:
Hi Johann,
I fixed the issue in new commit.
The use of jquery event namespace worked in release 4.3.1 but not in the master branch.
I replaced it with a named function.

XhmikosR commented:
@Johann-S: when you merge this please adapt it for v4 and push it to my v4-dev-xmr branch

Johann-S commented:
Now it's fixed thanks @david-lallement 👍 you can see it here: https://codepen.io/Johann-S/pen/zVGmzZ

and it'll be available in our next v4 release
